FEAT: Add a Message Manager for App

dipinknair opened the pull request:
Fixes #440 

```python
from ansys.mechanical.core import App
app =App()

# Add a custom message
app.messages.add("info", "User clicked the start button.")

# check number of messages
len(app.messages)
... 1

# returns panda data frame representation
app.messages
... Severity             TimeStamp                   DisplayString Source StringID                                Location RelatedObjects
... 0     INFO  1/30/2025 1:46:14 PM  User clicked the start button.   None     None  Ansys.ACT.Core.Utilities.SelectionInfo             []
# print minimalistic view (severity and message)
app.messages.show()
... Severity: info
... DisplayString: User clicked the start button.

# prints all information including location, time, source etc of all message available
app.messages.show(filter='*")
... TimeStamp: 1/30/2025 1:42:01 PM
... Severity: info
... DisplayString: User clicked the start button.
... Source: None
... StringID: None
... Location: Ansys.ACT.Core.Utilities.SelectionInfo
... RelatedObjects: System.Collections.Generic.List`1[Ansys.Mechanical.DataModel.Interfaces.IDataModelObject]

# gets specific message in datafram row representation
print(app.messages[1].Severity)
... Info

# Remove specific message
app.messages.remove(1)

# Clear all messages
app.messages.clear()
````

koubaa commented:
`messages.count()` isn't pythonic. It should be `len(messages)`

koubaa commented:
An on/off for complete_info seems inflexible. How about this:

`app.messages.show(filter="*") # filter defaults to "severity;message". Accepts a semicolon delimited list of columns or a wildcard.`

koubaa commented on src/ansys/mechanical/core/embedding/messages.py:
```diff
@@ -0,0 +1,220 @@
+
+    def show(self, complete_info=False):
+        """Print all messages with full details.
+
```
shouldn't this update the dataframe first?

dipinknair replied:
yes

dipinknair commented:
> An on/off for complete_info seems inflexible. How about this:
> 
> `app.messages.show(filter="*") # filter defaults to "severity;message". Accepts a semicolon delimited list of columns or a wildcard.`

Added this feature and updated the PR descriptions as well

koubaa commented on src/ansys/mechanical/core/embedding/app.py:
```diff
@@ -34,6 +34,7 @@
 from ansys.mechanical.core.embedding.addins import AddinConfiguration
 from ansys.mechanical.core.embedding.appdata import UniqueUserProfile
 from ansys.mechanical.core.embedding.imports import global_entry_points, global_variables
+from ansys.mechanical.core.embedding.messages import MessageManager
```
can you import this lazily? I think pandas does something on initializatino which might slow things down

koubaa commented on src/ansys/mechanical/core/embedding/app.py:
```diff
@@ -437,6 +439,14 @@ def project_directory(self):
         return self.DataModel.Project.ProjectDirectory
 
+    @property
+    def messages(self):
```
I don't like that `_update_messages_cache()` has to be done explicitly. The message manager should be responsible for keeping itself up to date. Use an event handler

koubaa commented on src/ansys/mechanical/core/embedding/messages.py:
```diff
@@ -0,0 +1,251 @@
+        self._update_messages_cache()
+
+
+class _Message:
```
why is this necessary? There is already a message object in the API

dipinknair replied:
We already have all messages stored as dataframe. Isnt it better to use that rather than doing API call for each message?

koubaa commented on src/ansys/mechanical/core/embedding/messages.py:
```diff
@@ -0,0 +1,190 @@
+        from Ansys.Mechanical.DataModel.Enums import MessageSeverityType
+
+        self._message_severity = MessageSeverityType
+        self._message = Message
```
self._message doesn't seem to be used

dipinknair replied:
self._message its used when adding new message

dipinknair commented:
@koubaa  When app starts there is already a warning message saying license checkout is delayed. This was causing all the tests to fail because how test is written assuming zero message at the beginning.
